chanzuckerberg/single-cell-data-portal
# feat(cellcard): 4937 GeneInfo Panel

## Reason for Change

- #4937

https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell-data-portal/4937

## Changes

1. Add GeneInfo Panel in Cell Card page
2. Extract WMG GeneInfoSideBar to `src/components/*`, so it's reused by both WMG and Cell Card
3. Extract WMG RightSideBar to `src/components/common/*`, so it's reused by both WMG and Cell Card
4. Update MarkerGeneTables to add gene info ButtonIcon and update `symbol` prop type from `string` to `ReactNode`
5. Add GeneInfoPanel to Cell Card

## Testing steps

RDEV: https://thuang-4937-geneinfo-frontend.rdev.single-cell.czi.technology/cell-guide/CL_0002306

1. Navigate to a Cell Card page
2. Scroll down to gene table
6. You'll see a new ButtonIcon next to each gene name
7. On click, it will open a right panel and load gene info
8. Click "View on GeneCards" link will take you to the GeneCards page for the gene
9. Click on "NCBI" link will take you to the NCBI gene page
10. Resize the page from wide to narrow will trigger `skinny mode` and hide in page navigation and move search bar to the top

https://github.com/chanzuckerberg/single-cell-data-portal/assets/6309723/3adcd01d-e64f-4eb6-b9fa-6cd3a9e91450



## Notes for Reviewer
